libbeat/cmd/instance/locks/process_lookup_stub.go

```diff
@@ -0,0 +1,31 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build (!darwin || !cgo) && !freebsd && !linux && !windows && !aix
```

Comment:
What are the implications of this failure? Do we need to implement a recovery path that does not rely on process state for non-CGO darwin builds? (I think APM is doing some work to ensure they could run without CGO on darwin so this might be something that's of interest to them. I'm not sure if their use was related to apm-server though.)

TestLockWithDeadPid fails for `CGO_ENABLED=0 GOOS=darwin`. So I think the tests that are dependent upon cgo must be moved to their own file that is guarded by build tags.

> cannot obtain lockfile: error looking up status for pid 99999: findMatchingPID not supported on platform: darwin


 //cc: @simitt

Reply:
Yah these weird wrapper files only exist because github's built-in CI will fail if it calls a function that depends on CGO. I would prefer not to have them at all, but the darwin code needs cgo for process lookup APIs.

I'm also not sure how we'd get this error under "real world" circumstances, as my understanding (and some tests I ran) was that everything is packaged with CGO on.